### PR TITLE
HID-2126 Tailwind CSS framework and Cube CSS

### PR DESCRIPTION
**Pull request by left23**
Branches off #160 which has form elements to test Tailwind css classes

1. Tailwind styles to all pages and settings tabs.
2. Cube CSS for the flow, to control vertical rhythm on pages and within the form elements
3. disabled all previous framework styles
4. cd-button component for the buttons
5. commented out the settings tabs as in the dev branch so only functional code is displayed

**Tailwind**

I've installed the library based on https://tailwindcss.com/docs/installation
And based on looking at all the examples https://github.com/tailwindlabs/tailwindcss-setup-examples
and googling, the css is being generated and purged as part of the start script. 

We load the whole library for development (~ 2.3mB) and then rely on purge to remove most rules when we generate the tailwind-styles.css file for production (in my example, the CSS has only a handful of rules and is 14kB) https://tailwindcss.com/docs/controlling-file-size

I've basic configuration in place
https://tailwindcss.com/docs/configuration

https://tailwindcss.com/docs/adding-base-styles
I've done a simple example of extending the rules in `tailwind-source.css`, for the headings.
The preflight https://tailwindcss.com/docs/preflight unstyles headings https://tailwindcss.com/docs/preflight#headings-are-unstyled 

Screenshot from 2020-11-04 14-38-52

### CODE REVIEW

**Comment by rupl** [comment]
Hey that's looking slick already!

File: tailwind.config.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module.exports = {
+  future: {
+    // removeDeprecatedGapUtilities: true,
+    // purgeLayersByDefault: true,
+  },
+  purge: {
+    enable: true, // To test locally.
```
Comment (left23) [inline tailwind.config.js:7]: @rupl This didn't work for me.
With this set to true, I expected the purge to run regardless of environment, but it took https://github.com/UN-OCHA/hid_api/pull/161/commits/b28c66d1a7c5d0c0e976eadcce94a0be20c1520a#diff-1671da1302cd5c094dd762e76e142d8c1caf8f276eb884b86ef1630507e81ed2 `NODE_ENV=production ` to get the CSS purge happening locally for me.

File: package.json [hunk 2]
```diff
@@ -12,10 +12,11 @@
   "keywords": [],
   "main": "index.js",
   "scripts": {
-    "dev": "node ./node_modules/nodemon/bin/nodemon.js ./server.js",
+    "dev": "npm run build:tailwind && node ./node_modules/nodemon/bin/nodemon.js ./server.js -e ejs,js,css,html",
```
Comment (left23) [inline package.json:15]: @rupl I'm running `docker-compose up` locally and I can see it runs the start script. Should I be running the `dev` script locally? Then I could avoid adding the `NODE_ENV=production` in `start`?
To be discusses whenever. For now, this all works!

**Comment by left23** [comment]
@rupl Maybe more discussion is needed before we commit to this, but in the meantime, feel free to casually review. I don't consider this a candidate for merging yet.

**Comment by left23** [comment]
@rupl The latest changes apply tailwind styles to all pages and settings tabs. 
I've introduced Cube CSS for the flow only, to control vertical rhythm on pages and within the form elements.
I've disabled all bootstrap-related styles (or previous framework, not sure it's BS) so this should expose any non-styled elements remaining, although I hope I caught them all and replaced them with tailwind.
I'm also using the cd-button component for the buttons to test its flexibility. If this proves insufficient, I think we can look at using tailwind for the buttons too.
Lastly, I commented out the settings tabs as in the dev branch so only functional code is displayed. If you do review this from a visual perspective, you'll need to uncomment a lot of the markup in settings.html
I'm parking this until we get a chance to review together. I think tailwind is helpful for the layout and form elements but I'm curious to hear your thoughts on whether it's worth it and/or something you picture keeping.

File: package.json (hunk 2 above)
Comment (rupl) [reply package.json:15]: I've opened a ticket to work on a fix for this: https://humanitarian.atlassian.net/browse/OPS-7176

File: tailwind.config.js (hunk 1 above)
Comment (rupl) [reply tailwind.config.js:7]: I agree your config is the sensible thing here. We'll see if we can solve this once we have resolved https://humanitarian.atlassian.net/browse/OPS-7176

File: tailwind.config.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      keyframes: true,
+    },
+  },
+  theme: {
+    extend: {
+      outline: {
+        ocha: ['2px solid var(--cd-bright-blue)', '1px'],
+      },
+    },
+  },
```
Comment (rupl) [inline tailwind.config.js:20]: per our conversation, `theme` can be cut as our use-case doesn't seem to require this complexity

**Comment by rupl** [comment]
Closing this based on discussion in #170 and in Flowdock. We captured most of it in the other PR and anything we missed can be addressed in follow-ups.

**Pull request closed without merging**